### ISSUE TITLE
Props not getting passed down to child SVG elements properly?

### ISSUE THREAD

**Issue by dylancristy** [drcmda/react-spring#90]
I am trying to animate some "buttons" made with SVG elements, and it seems if I wrap the entire construct in a Spring, some of the props are not passed down to the nested SVG elements correctly.  I've set up the scenario here:

https://codesandbox.io/s/7wo28qm6k0

Please refer first to the NavButtonWorking.js file.  In the example that first comes up, I've wrapped one of the inner `circle` elements in a Spring to animate `stroke` and `stroke-dashoffset`.  The circle is getting it's `cx`, `cy`, `r` and `strokeWidth` attribute values from props passed to the parent component, and I think it's working in this situation because I've exposed those props as `const`s inside the render function, so they are somehow available in scope.

Now refer to the NavButtonBroken.js file.  I've removed the SVG structure and made it into it's own function which gets props passed in, and I've changed the outer `svg` element into an `animated.svg` because I want to animate the scale of the entire thing. (This structure of the file is based on your example [here](https://codesandbox.io/embed/882njxpz29) where you are passing props down into nested SVG elements.)

Now, if you go to the LargeButton.js file and change which inner NavButton component it's using to be the broken one, you'll see the buttons disappear.  But, if you inspect the page, you'll see that the `svg` and `circle` elements are there... they just don't have their `cx`, `cy`, `r` or `stroke-width` attributes, which should have been set by the props, so they have no dimensional information which I'm guessing is why they don't get rendered.

The only place I see where the props got passed and used is in the outermost `svg` which has it's `scale` attribute set properly, and in the `style` attribute of the last `circle` element, which is the one I am trying to animate `stroke` and `stroke-dashoffset` for, and those values are being set through the `style` attribute and not direct attributes on the element.

So.... am I doing something wrong here?  Is there a different way to pass down props?  I'm even trying to pass down a click handler and attach it to a `g` element, just as you do in your example I linked to, but in mine the handler doesn't attach.

**Comment by drcmda** [comment]
@dylancristy I looked over it but couldn't see anything wrong stemming from r-spring, to me the structure itself looks odd though, you're putting a svg into a svg, is that permitted?

**Comment by dylancristy** [comment]
@drcmda  Yes, nested `svg`s are permitted, although I may be doing it for the wrong reason.

In any case, take a look at two more files I've added.

In NavButtonThree.js, I took out the `Spring` component but left the inner `Button` as a function, to see if there was anything wrong with having that as a function that receives props and tries to pass them down to inner elements.  That one works fine, all the props get passed down as expected and the buttons render as they should (although now there's no animation).  I did have to do something slightly different with the `onClick` there, but still, you can see that props get passed down correctly.

In NavButtonFour.js I added the `Spring` back in, and on a hunch, I explicitly added all the props I was trying to pass down to the inner elements to the `to` prop of the `Spring` component, even though those props aren't getting animated.  That didn't work either until I made _all_ the inner elements the `animated` versions of themselves (except for the `g`, for some reason that one can still be a regular `g`).  And even that is not working completely.  It works for the `circle` and `text` elements, but I'm trying to pass a couple values as props to go into the `filter` element, and once I try to make those into `animated` versions, things just fall apart.

So, is it the case that you have to explicitly include all props you want passed down in the `to` prop of the `Spring`?  That seems... inconvenient.  And also, is it true that you can't mix and match regular elements and `animated` elements, even if some of the elements don't need animating?

My intuitive sense of how it should work would be that:

- Any elements you want animated should be the `animated` version, but elements that are not animating can be regular elements.

- The only things you need to include in the `to` prop of the `Spring` are the things you want animated, all other props should be passed down transparently.

Is it not working that way by design?

I also just added a third new file NavButtonFive.js, in which I removed the `native` prop from the `Spring` component, and made all the inner `svg` elements regular.  Again, this only works if I explicitly pass in all the props in the `to` prop of the `Spring`.

**Comment by dylancristy** [comment]
Aha!  Gotcha, thanks!

I guess I misinterpreted the sentence "we offer...prop forwarding (unrecognized props will be spread over the receiving component)" to mean that I didn't have to do that explicitly myself.
